include/seqan3/search/algorithm/detail/search_scheme_algorithm.hpp

```diff
@@ -115,7 +115,7 @@ inline auto search_scheme_block_info(search_scheme_t const & search_scheme, size
 
     uint8_t const blocks      {search_scheme[0].blocks()};
     size_t  const block_length{query_length / blocks};
-    uint8_t const rest        {query_length - blocks * block_length};
+    uint8_t const rest        {static_cast<uint8_t>(query_length % blocks)};
```

Comment:
Are you sure we want to do this? `mod` has the complexity of `div` which should be much slower than the multiplication and substraction (since we already did the `div` in the line above)

Reply:
A division always delivers the rest on intel cpu's for free. See https://godbolt.org/z/KqID2a.

So you make it only more difficult for the compiler to see this.